libs/mrwebrtc/src/media/external_video_track_source.h

```diff
@@ -170,14 +171,17 @@ class ExternalVideoTrackSource : public TrackedObject,
   /// Shutdown the source and release the buffer adapter and its callback.
   void Shutdown() noexcept;
 
-  webrtc::VideoTrackSourceInterface* impl() const { return track_source_; }
-
  protected:
-  ExternalVideoTrackSource(RefPtr<GlobalFactory> global_factory,
-                           std::unique_ptr<detail::BufferAdapter> adapter);
+  ExternalVideoTrackSource(
+      RefPtr<GlobalFactory> global_factory,
+      std::unique_ptr<detail::BufferAdapter> adapter,
+      rtc::scoped_refptr<detail::CustomTrackSourceAdapter> source);
   // void Run(rtc::Thread* thread) override;
   void OnMessage(rtc::Message* message) override;
 
+  // Keep a second reference (in addition of VideoTrackSource::source_) as a
+  // pointer to the specific type of adapter source, to simplify access and
+  // avoid having to cast everywhere.
   rtc::scoped_refptr<detail::CustomTrackSourceAdapter> track_source_;
```

Comment:
nit: can we have a helper function do the cast rather than storing a separate pointer? Just to avoid pointers getting out of sync.

Reply:
Fixed.